HIP/hip-1056.md

```diff
@@ -0,0 +1,1795 @@
+  necessary for performance reasons or if rounds become very frequent and small.
+    - A block *may* be zero rounds because the last block prior to a network "freeze" must communicate the state hash
+      (via a block proof) of the state immediately prior to network shutdown. This is only possible with an extra empty
+      block because the state hash in each block proof is the state hash at the *beginning* of that block/round. This
+      delay of state hash communication is required to simultaneously meet Hedera performance goals and also meet EVM
+      expectations for immediate access to recent (the last 256) block hash values.
+- A `Round` contains a sequence of zero or more `Events` that have reached consensus at the same time. Round length in real
+  world time depends on internet latency and event creation rate. In Hedera mainnet today it is around 1 round per
+  second. In the future this could be shorter to lower latency, but would never make sense to be less than ping time.
+- An `Event` contains a sequence of zero or more `EventTransactions` submitted by a single node.
+- Each `EventTransaction` is either a `SignedTransaction` or a system transaction.
+- A `SignedTransaction` is transmitted as a byte array, contains the signed transaction bytes of a user-submitted
+  transaction exactly as processed by the network, and is followed by transaction outputs and state changes resulting
+  from that transaction.
+- System transactions are represented by explicitly defined protobuf messages and used for consensus node to consensus
+  node communication.
+
+These concepts are core to the Hashgraph consensus mechanism and have direct impacts on the structure of data in the 
+streams.
+
+
+### Important Note Regarding Map Fields
+
+The network will never add a `map` to a protobuf message in the Block Stream, and will not use a `map` for any value in
+state. The protocol buffer implementations across different languages do not, and cannot, guarantee iteration order for
+map entries and some implementations deliberately randomize the iteration order to "aggressively" prevent any code
+from depending on map iteration order.
+
+This unreliable iteration order has the consequence that block validation would fail non-deterministically when
+encountering a different iteration order because the serialized bytes would not be in the same sequence and would not
+generate the same hash. All implementations of block validation would be required to use custom protobuf libraries,
+which would be required to use *the same* deterministic iteration order when serializing bytes, so that the generated
+hash value would match every other implementation. The network values the freedom to implement a compliant Block Stream
+consumer using any valid protocol buffer library and any computing language of choice, and therefore chooses not to use
+the `map` type in protobuf messages.
```

Comment:
Speaking of custom protobuf implementation, it may be worth mentioning that `pbj` actually has support for maps that guarantees ordering (see https://github.com/hashgraph/pbj/pull/258).

Reply:
Everything in block stream has to work with other protobuf implementations (e.g. protoc or tonic) as well; which means we cannot use maps in Block Streams.